lib/plug_datadog_stats.ex

```diff
@@ -1,43 +1,62 @@
 defmodule PlugDatadogStats do
   @behaviour Plug
-  import Plug.Conn, only: [register_before_send: 2]
+  import Plug.Conn, only: [register_before_send: 2, put_private: 3]
   require Logger
 
-  def init(_) do
-    %{
-      histogram: Application.get_env(:plug_datadog_stats, :histogram_name, "resp_time"),
-      count: Application.get_env(:plug_datadog_stats, :count_name, "resp_count"),
-    }
+  @base_opts %{
+    histogram: Application.get_env(:plug_datadog_stats, :histogram_name, "resp_time"),
+    count: Application.get_env(:plug_datadog_stats, :count_name, "resp_count"),
+    path_normalize_fn: {__MODULE__, :path_override_passthrough, []}
+  }
+
+  def init(opts \\ []) do
+    opts
+    |> Enum.into(@base_opts)
+    |> parse_options()
   end
 
-  def call(conn, names) do
+  def call(conn, opts) do
     req_start_time = :os.timestamp
 
     register_before_send conn, fn conn ->
-      tags = tags_for_conn(conn)
 
       # This will log response time in microseconds
       req_end_time = :os.timestamp
       duration = :timer.now_diff(req_end_time, req_start_time)
 
-      Logger.debug("PlugDatadogStats: #{duration}µs #{names.histogram}/#{names.count} #{inspect tags}")
-      ExStatsD.histogram(duration, names.histogram, tags: tags)
-      ExStatsD.increment(names.count, tags: tags)
+      tags = tags_for_conn(conn, opts.path_normalize_fn)
+
+      Logger.debug("PlugDatadogStats: #{duration}µs #{opts.histogram}/#{opts.count} #{inspect tags}")
+      ExStatsD.histogram(duration, opts.histogram, tags: tags)
+      ExStatsD.increment(opts.count, tags: tags)
 
-      conn
+      put_private(conn, :plug_datadog_stats_tags, tags)
     end
   end
 
-  defp tags_for_conn(conn) do
+  defp parse_options(%{path_normalize_fn: {_m, _f, _a}} = opts) do
+    opts
+  end
+
+  defp parse_options(%{path_normalize_fn: _bad}) do
+    raise "path_normalize_fn should be specified as {m, f, a}"
```

Comment:
I feel like if I saw this error I would have no idea what it means, can it be made a bit more explicit?

Reply:
`the path_normaliz_fn option should be provided in the format {module, function, arguments}.  Example: {MyModule, :function_name, []}`

Is that more understandable/user friendly?
`{m, f, a}` is a common method of passing methods around to be invoked later in Elixir.  In this case we are using it because an anonymous method can't be provided to an expression that's resolved at compile time.

The reason for that as I understand it is that since anon functions in Elixir are closures, they don't survive the transition because the compile time environment (to which they could still hold references) does not exist anymore.